### ISSUE TITLE
🐛 [firebase_core] Issues when updated from 1.18 to 1.19 on iOS

### ISSUE THREAD

**Issue by Fernandomr88** [firebase/flutterfire#9102]
## Bug report

**Describe the bug**

**This is happening only on iOS.**

Upgraded firebase_core on pubspec.yaml from 1.18.0 to 1.19.x and my app launched then the screen was blank.
The debugger printed this:
`[VERBOSE-2:ui_dart_state.cc(198)] Unhandled Exception: MissingPluginException(No implementation found for method Firebase#initializeCore on channel plugins.flutter.io/firebase_core)`
![image](https://user-images.githubusercontent.com/14304128/179067713-7c3b03ff-2431-4643-8e1a-8c81afa599ec.png)

Even when added 

```
 # Override Firebase SDK Version
$FirebaseSDKVersion = '8.15.0'
```

to my podfile it didnt work, same error, after pod updating.

Using flutter version 3.0.4 and now 3.0.5 happened the same thing.

**So when I changed back to `firebase_core: 1.18.0` (without the caret), it worked.**

### Steps to reproduce
just changed package version

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale pt-BR)
    • Flutter version 3.0.5 at /Users/fernandorocha/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (25 hours ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/fernandorocha/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.69.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • iPhone 11 (mobile) • DD7EA6B7-F804-4D92-99EE-EAE47DEAFC53 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.6
Flutter SDK 3.0.5
f****** 0.0.1+14

dependencies:
- amplify_auth_cognito 0.6.1 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_core aws_common collection flutter meta plugin_platform_interface]
- amplify_flutter 0.6.1 [amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios aws_common collection flutter meta plugin_platform_interface]
- async 2.9.0 [collection meta]
- brasil_fields 1.4.5 [flutter intl]
- camera 0.10.0 [camera_android camera_avfoundation camera_platform_interface camera_web flutter flutter_plugin_android_lifecycle quiver]
- cloud_functions 3.3.0 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- cupertino_icons 1.0.5
- currency_text_input_formatter 2.1.7 [flutter intl]
- datetime_picker_formfield 2.0.0 [flutter intl]
- dropdown_plus 0.0.9 [flutter]
- file_picker 4.6.1 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path win32]
- firebase_analytics 9.1.10 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.18.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 2.8.2 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_performance 0.8.0+14 [firebase_core firebase_core_platform_interface firebase_performance_platform_interface firebase_performance_web flutter]
- firebase_remote_config 2.0.10 [firebase_core firebase_core_platform_interface firebase_remote_config_platform_interface firebase_remote_config_web flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_credit_card 3.0.1 [flutter]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_pdfview 1.2.2 [flutter]
- flutter_speed_dial 6.0.0 [flutter]
- flutter_statusbarcolor_ns 0.4.0 [flutter]
- fluttertoast 8.0.9 [flutter flutter_web_plugins]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- launch_review 3.0.1 [flutter]
- open_file 3.2.1 [flutter ffi]
- package_info_plus 1.4.2 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- path 1.8.1
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- share_plus 4.0.10 [meta mime flutter share_plus_platform_interface share_plus_linux share_plus_macos share_plus_windows share_plus_web]
- url_launcher 6.1.5 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

dev dependencies:
- flutter_launcher_icons 0.9.3 [args image path yaml]
- flutter_lints 2.0.1 [lints]

transitive dependencies:
- amplify_auth_cognito_android 0.6.1 [flutter]
- amplify_auth_cognito_ios 0.6.1 [amplify_core flutter]
- amplify_core 0.6.1 [aws_common collection flutter intl json_annotation meta plugin_platform_interface uuid]
- amplify_datastore_plugin_interface 0.6.1 [amplify_core collection flutter meta]
- amplify_flutter_android 0.6.1 [flutter]
- amplify_flutter_ios 0.6.1 [amplify_core flutter]
- archive 3.3.0 [crypto path]
- args 2.3.1
- aws_common 0.1.1 [async collection http meta stream_transform uuid]
- camera_android 0.10.0 [camera_platform_interface flutter flutter_plugin_android_lifecycle stream_transform]
- camera_avfoundation 0.9.8+2 [camera_platform_interface flutter stream_transform]
- camera_platform_interface 2.2.0 [cross_file flutter plugin_platform_interface stream_transform]
- camera_web 0.3.0 [camera_platform_interface flutter flutter_web_plugins stream_transform]
- characters 1.2.0
- clock 1.1.0
- cloud_functions_platform_interface 5.1.10 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.2.17 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.16.0
- cross_file 0.3.3+1 [js meta]
- crypto 3.0.2 [typed_data]
- ffi 1.2.1
- file 6.1.2 [meta path]
- firebase 9.0.3 [http http_parser js]
- firebase_analytics_platform_interface 3.1.10 [firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.4.0+16 [firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_core_platform_interface 4.4.1 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.6.5 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.2.10 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_performance_platform_interface 0.1.1+10 [firebase_core flutter plugin_platform_interface]
- firebase_performance_web 0.1.0+15 [firebase firebase_core firebase_core_web firebase_performance_platform_interface flutter flutter_web_plugins js]
- firebase_remote_config_platform_interface 1.1.10 [firebase_core flutter meta plugin_platform_interface]
- firebase_remote_config_web 1.0.15 [firebase_core firebase_core_web firebase_remote_config_platform_interface flutter flutter_web_plugins js]
- flutter_plugin_android_lifecycle 2.0.6 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- image 3.2.0 [archive meta xml]
- js 0.6.4
- json_annotation 4.6.0 [meta]
- lints 2.0.0
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- mime 1.0.2
- package_info_plus_linux 1.0.5 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.5 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.5 [package_info_plus_platform_interface ffi flutter win32]
- path_provider_android 2.0.16 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.10 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.4 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.7 [ffi flutter path path_provider_platform_interface win32]
- petitparser 5.0.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- process 4.2.4 [file path platform]
- quiver 3.1.0 [matcher]
- share_plus_linux 3.0.0 [share_plus_platform_interface file flutter meta url_launcher]
- share_plus_macos 3.0.1 [share_plus_platform_interface flutter]
- share_plus_platform_interface 3.0.3 [flutter meta mime plugin_platform_interface]
- share_plus_web 3.0.1 [share_plus_platform_interface url_launcher flutter flutter_web_plugins meta]
- share_plus_windows 3.0.1 [share_plus_platform_interface flutter meta url_launcher]
- sky_engine 0.0.99
- source_span 1.9.0 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_transform 2.0.0
- string_scanner 1.1.1 [source_span]
- term_glyph 1.2.0
- typed_data 1.3.1 [collection]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.0 [flutter plugin_platform_interface]
- url_launcher_web 2.0.12 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
- win32 2.6.1 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 6.1.0 [collection meta petitparser]
- yaml 3.1.1 [collection source_span string_scanner]
```

</details>

---

**Comment by Fernandomr88** [comment]
> Thanks for the report @Fernandomr88
> 
> in podfile, can you add below line and see if it works ?
> 
> `pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git[](https://github.com/invertase/firestore-ios-sdk-frameworks.git)', :branch => 'main'`
> 
> also, try running below commands and confirm:
> 
> ```
> rm -Rf ios/Pods
> rm -Rf ios/.symlink
> rm -Rf ios/Flutter/Flutter.framework
> rm -Rf ios/Flutter/Flutter.podspec
> rm -rf ios/Podfile.lock
> rm -rf ~/Library/Developer/Xcode/DerivedData/* -y
> rm -rf pubspec.lock
> flutter clean
> flutter packages get
> cd ios
> pod install
> pod update
> ```

That repository you suggested is from a random user in github, why would I use that? is it safe?
My intention with this issue was that this package (firebase_core) needs fixing, in the meantime i'm using firebase_core 1.18.0

**Comment by darshankawar** [comment]
> That repository you suggested is from a random user in github, why would I use that? is it safe?

Thanks for feedback. It is actually not a random user's github, but of `invertase` org who is developing and maintaining flutterfire plugins. 



> My intention with this issue was that this package (firebase_core) needs fixing, in the meantime i'm using firebase_core 1.18.0

I tried running the `firebase_core` plugin example on iOS with latest version (plugin) but it didn't give me the reported error.

Can you provide us the complete error log ? Do you have any other third party plugin which could be conflicting with `firebase_core` ?

**Comment by Fernandomr88** [comment]
fixed with version 1.20
